go/login/server/server.go

```diff
@@ -0,0 +1,181 @@
+)
+
+var (
+	// ErrInvalidResponseLen denotes that response length provided is invalid
+	ErrInvalidResponseLen = fmt.Errorf("invalid response length provided")
+)
+
+// Authorizer responds to an authorization request. The method
+// GrantLogin returns whether an user is allowed to run an action in a host, and a error value.
+//
+// Some considerations need to be held while implementing this interface:
+// - Allow should consider that an empty string as command is a correct input.
+// - If no user can be obtained from request metadata, an empty string is to be
+// passed as default value.
+// - Both hostIDType and hostID can be empty. Whether this refer to a default value
+// or not is to be user configurable.
+// - returned boolean will be considered even if an error is returned.
+type Authorizer interface {
+	GrantLogin(user string, hostID string, hostIDType string, action string) (bool, error)
+}
+
+// AuthorizerFunc type is an adapter to allow the use of ordinary functions as an Authorizer.
+type AuthorizerFunc func(user string, hostID string, hostIDType string, action string) (bool, error)
+
+// GrantLogin calls a(user, hostID, hostIDType, action)
+func (a AuthorizerFunc) GrantLogin(user string, hostID string, hostIDType string, action string) (bool, error) {
+	return a(user, hostID, hostIDType, action)
+}
+
+// LoginServer is a framework for glome-login servers, Creation of the object is managed via
+// the NewLoginServer method. To create one it is necessary to provide an authorizer. Optionally, a series
+// of options can be added at creation.
+type LoginServer struct {
+	// Keys manages the keys used by the server.
+	Keys *KeyManager
+
+	auth        Authorizer
+	authLock    sync.RWMutex
+	loginParser *login.Server
+
+	responseLen uint8
+	userHeader  string
+}
+
+//  GrantLogin implements the Authorizer interface for AuthorizerFunc by calling the wrapped function.
+func (s *LoginServer) Authorizer(a Authorizer) {
+	s.authLock.Lock()
+	s.auth = a
+	s.authLock.Unlock()
+}
+
+// NewLoginServer creates a new server with provided Authorizer and, optionally, selected options.
+func NewLoginServer(a Authorizer, options ...func(*LoginServer) error) (*LoginServer, error) {
+	srv := LoginServer{
+		auth:        a,
+		Keys:        NewKeyManager(),
+		responseLen: MaxResponseSize,
+		userHeader:  "authenticated-user",
+	}
+	srv.loginParser = srv.newLoginParser()
+
+	for _, option := range options {
+		if err := option(&srv); err != nil {
+			return nil, err
+		}
+	}
+
+	return &srv, nil
+}
+
+// ResponseLen is an option to be provided to NewServer on creation. Its sets the size of response
+// to provided length. the size is measured in number of characters in base64. Raise error if provided
+// length is not in 0-MaxResponseSize. If not set, defaults to MaxResponseSize.
+func ResponseLen(length uint8) func(srv *LoginServer) error {
+	return func(srv *LoginServer) error {
+		if !(0 < length && length <= MaxResponseSize) {
```

Comment:
```suggestion
		if length <= 0 || length > MaxResponseSize) {
```

I consider that easier to read, pushing the negation into the term.